wiki.moinmoin

```diff
@@ -380,15 +379,51 @@ RULE:     - the owning team will provide timely, high quality updates for the
 RULE:       security team to sponsor to fix issues in the affected vendored code
 RULE:     - if subsequent uploads add new vendored components or dependencies
 RULE:       these have to be reviewed and agreed by the security team.
+RULE:     - Such updates in the project might be trivial, but imply that a
+RULE:       dependency for e.g. a CVE fix will be moved to a new major version.
+RULE:       Being vendored that does gladly at least not imply incompatibility
+RULE:       issues with other packages or the SRU policy. But it might happen
+RULE:       that this triggers either:
+RULE:       a) The need to adapt the current version of the main package and/or
+RULE:          other vendored dependencies to work with the new dependency
+RULE:       b) The need to backport the fix in the dependency as the main
+RULE:          package will functionally only work well with the older version
+RULE:       c) The need to backport the fix in the dependency, as it would imply
+RULE:          requiring a newer toolchain to be buildable that isn't available
+RULE:          in the target release.
+RULE: - The rust ecosystem currently isn't yet considered stable enough for
+RULE:   classic lib dependencies and transitions in main; therefore the
+RULE:   expectation for those packages is to vendor (and own/test) all
+RULE:   dependencies (except those provided by the rust runtime itself).
+RULE:   This implies that all the rules for vendored builds always
+RULE:   apply to them. In addition
+RULE:   - Rust builds using librust-*-dev packges will populate the attribute
+RULE:     `X-Cargo-Built-Using`, right now this is expected not to be present.
+RULE:   - It is expected rust builds will use dh-cargo so that a later switch
+RULE:     to proper package dependencies isn't too hard (e.g. it is likely
+RULE:     that over time some more common/stable libs shall come from proper
+RULE:     archive packages).
+RULE: - All vendored dependencies (no matter what language) shall have a
+RULE:   way to be refreshed (recommended is a documentation README.source and
+RULE:   scripts in debian/
```

Comment:
I think this is more code/language-base knowledge than anything else. I don’t know for rust, but for golang, all projects using modules (which is the default now) should have the same files to edit and strategy. Should the project has a file "this is a standard go projects using vendored modules?" (then, those using other tools like godep can link how to use them as it’s not part of the base tooling)

Reply:
You are right, I have updated this section after I have had some TIL got go-mod and cargo-update :-)
Thanks for bringing it up.